python/orca/tutorial/NCF/pytorch_resume_spark_dataframe.py

```diff
@@ -0,0 +1,111 @@
+from bigdl.orca.learn.metrics import Accuracy, Precision, Recall
+
+
+# Step 1: Init Orca Context
+args = parse_args("PyTorch NCF Resume Training with Spark DataFrame")
+init_orca(args, extra_python_lib="pytorch_model.py")
+spark = OrcaContext.get_spark_session()
+
+
+# Step 2: Load the processed
+train_df = spark.read.parquet(os.path.join(args.data_dir, "train_processed_dataframe.parquet"))
+test_df = spark.read.parquet(os.path.join(args.data_dir, "test_processed_dataframe.parquet"))
+
+
+# Step 3: Define the model, optimizer and loss
+def model_creator(config):
+    model = NCF(user_num=config["user_num"],
+                item_num=config["item_num"],
+                factor_num=config["factor_num"],
+                num_layers=config["num_layers"],
+                dropout=config["dropout"],
+                model=config["model"],
+                sparse_feats_input_dims=config["sparse_feats_input_dims"],
+                sparse_feats_embed_dims=config["sparse_feats_embed_dims"],
+                num_dense_feats=config["num_dense_feats"])
+    model.train()
+    return model
+
+
+def optimizer_creator(model, config):
+    return optim.Adam(model.parameters(), lr=config["lr"])
+
+
+def scheduler_creator(optimizer, config):
+    scheduler = StepLR(optimizer, step_size=1)
+    return scheduler
+
+loss = nn.BCEWithLogitsLoss()
+
+
+# Step 4: Resume distributed training with Orca PyTorch Estimator
+config = load_model_config(args.model_dir, "config.json")
+callbacks = [TensorBoardCallback(log_dir=os.path.join(args.model_dir, "logs"),
+                                 freq=1000)] if args.tensorboard else []
+scheduler = scheduler_creator if args.lr_scheduler else None
+
+est = Estimator.from_torch(model=model_creator,
+                           optimizer=optimizer_creator,
+                           scheduler_creator=scheduler,
+                           loss=loss,
+                           metrics=[Accuracy(), Precision(), Recall()],
+                           backend=args.backend,
+                           use_tqdm=True,
+                           workers_per_node=args.workers_per_node,
+                           config=config)
+est.load(os.path.join(args.model_dir, "NCF_model"))
+train_stats = est.fit(data=train_df, epochs=2,
+                      feature_cols=feature_cols,
+                      label_cols=label_cols,
+                      batch_size=10240,
+                      validation_data=test_df,
+                      callbacks=callbacks)
+print("Train results:")
+for epoch_stats in train_stats:
+    for k, v in epoch_stats.items():
+        print("{}: {}".format(k, v))
+    print()
+
+
+# Step 5: Distributed evaluation of the trained model
+eval_stats = est.evaluate(data=test_df,
+                          feature_cols=feature_cols,
+                          label_cols=label_cols,
+                          batch_size=10240)
+print("Evaluation results:")
+for k, v in eval_stats.items():
+    print("{}: {}".format(k, v))
```

Comment:
For resume training, I think we can remove the evaluate here, to make it simple.